### PR TITLE
[KubeRay][Doc] KubeRay doc migration (1/n)

### PR DESCRIPTION
**Pull request by kevin85421**
## Related issue number

I failed to create a child page under "Managed Kubernetes Services." I believe there might be some layer limitations in our documentation. The [VM launcher](https://github.com/ray-project/ray/blob/dea8ad7cb7dd98eb5a0d75a053847ce282f84230/doc/source/_toc.yml#L346-L351) also tried to do similar things, but I can't see an entry like "Deploying on YARN" in the sidebar.

Screen Shot 2023-08-21 at 4 06 26 PM

wants to add the doc in a deeper layer, but  

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by kevin85421** [comment]
cc @angelinalg would you mind reviewing this PR? Btw, I am not familiar with Ray docs. Are there any tests that I need to run in advance locally? Thanks!

**Review by architkulkarni — APPROVED** [review]
Stamping in case `cluster` codeowner approval is required.

File: doc/source/cluster/kubernetes/examples/mobilenet-rayservice.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/examples/mobilenet-rayservice.md:31]:
```suggestion
## Step 4: Forward the port for Ray Serve
```

File: doc/source/cluster/kubernetes/examples/mobilenet-rayservice.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/examples/mobilenet-rayservice.md:37]:
```suggestion
Note that the Ray Serve service is created after the Ray Serve applications are ready and running. This process may take approximately 1 minute after all Pods in the RayCluster are running.
```

File: doc/source/cluster/kubernetes/examples/mobilenet-rayservice.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/examples/mobilenet-rayservice.md:21]:
```suggestion

```

File: doc/source/cluster/kubernetes/examples/mobilenet-rayservice.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/examples/mobilenet-rayservice.md:19]:
```suggestion

* Step 3.1: Download `ray-service.mobilenet.yaml`
```

File: doc/source/cluster/kubernetes/examples/mobilenet-rayservice.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/examples/mobilenet-rayservice.md:24]:
```suggestion
* Step 3.2: Create a RayService
```sh
```

File: doc/source/cluster/kubernetes/examples/mobilenet-rayservice.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/examples/mobilenet-rayservice.md:22]:
```suggestion
curl -LO https://raw.githubusercontent.com/ray-project/kuberay/master/ray-operator/config/samples/ray-service.mobilenet.yaml
```
add ```

File: doc/source/cluster/kubernetes/getting-started/raycluster-quick-start.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/getting-started/raycluster-quick-start.md:10]:
```suggestion
* Make sure your Kubernetes cluster has at least 4 CPU and 4 GB RAM.
```

File: doc/source/cluster/kubernetes/getting-started/rayjob-quick-start.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/getting-started/rayjob-quick-start.md:5]:
```suggestion
> Note: Ray Job Support in KubeRay is in alpha. Ray Job improvements are planned for future releases.
```

File: doc/source/cluster/kubernetes/getting-started/rayjob-quick-start.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/getting-started/rayjob-quick-start.md:14]:
```suggestion
A RayJob manages two aspects:
```

File: doc/source/cluster/kubernetes/getting-started/rayservice-quick-start.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/getting-started/rayservice-quick-start.md:2]:
```suggestion
This guide focuses solely on the Ray Serve multi-application API, which is available starting from Ray version 2.4.0.
```

File: doc/source/cluster/kubernetes/getting-started/rayservice-quick-start.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/getting-started/rayservice-quick-start.md:7]:
```suggestion
```

File: doc/source/cluster/kubernetes/getting-started/rayservice-quick-start.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/getting-started/rayservice-quick-start.md:14]:
```suggestion
A RayService manages 2 aspects:
```

File: doc/source/cluster/kubernetes/getting-started/rayjob-quick-start.md
Comment (angelinalg) [reply doc/source/cluster/kubernetes/getting-started/rayjob-quick-start.md:5]: Suggest using the {warning} component like the one on the landing page of Ray Serve (which has  {tip} component).

File: doc/source/cluster/kubernetes/index.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/index.md:23]:
```suggestion
These CRDs assist users in efficiently managing Ray clusters tailored to various use cases.
```

File: doc/source/cluster/kubernetes/index.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/index.md:25]:
```suggestion
See [Getting Started](kuberay-quickstart) to learn the basics of KubeRay and follow the quickstart guides to run your first Ray application on Kubernetes with KubeRay.
```

File: doc/source/cluster/kubernetes/index.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/index.md:90]:
```suggestion
        Integrate KubeRay with third party Kubernetes ecosystem tools.
```

File: doc/source/cluster/kubernetes/index.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/index.md:113]:
```suggestion
            Benchmark results
```

File: doc/source/cluster/kubernetes/index.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/index.md:120]:
```suggestion
        Consult the KubeRay troubleshooting guides.
```

File: doc/source/cluster/kubernetes/index.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/index.md:128]:
```suggestion
            Troubleshooting guides
```

File: doc/source/cluster/kubernetes/index.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/index.md:98]:
```suggestion
            Ecosystem Guides
```

File: doc/source/cluster/kubernetes/k8s-ecosystem/ingress.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/k8s-ecosystem/ingress.md:5]:
```suggestion
Two examples show how to use ingress to access your Ray cluster:
```

File: doc/source/cluster/kubernetes/k8s-ecosystem/ingress.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/k8s-ecosystem/ingress.md:7]:
```suggestion
  * [AWS Application Load Balancer (ALB) Ingress support on AWS EKS](#example-aws-application-load-balancer-alb-ingress-support-on-aws-eks)
```

File: doc/source/cluster/kubernetes/k8s-ecosystem/ingress.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/k8s-ecosystem/ingress.md:8]:
```suggestion
  * [Manually setting up NGINX Ingress on KinD](#example-manually-setting-up-nginx-ingress-on-kind)
```

File: doc/source/cluster/kubernetes/k8s-ecosystem/ingress.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/k8s-ecosystem/ingress.md:10]:
```suggestion
## AWS Application Load Balancer (ALB) Ingress support on AWS EKS
```

File: doc/source/cluster/kubernetes/k8s-ecosystem/ingress.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/k8s-ecosystem/ingress.md:74]:
```suggestion
## Manually setting up NGINX Ingress on KinD
```

File: doc/source/cluster/kubernetes/k8s-ecosystem/ingress.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/k8s-ecosystem/ingress.md:13]:
```suggestion
* Create an EKS cluster. See [Getting started with Amazon EKS – AWS Management Console and AWS CLI](https://docs.aws.amazon.com/eks/latest/userguide/getting-started-console.html#eks-configure-kubectl).
```

File: doc/source/cluster/kubernetes/k8s-ecosystem/ingress.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/k8s-ecosystem/ingress.md:15]:
```suggestion
* set up the [AWS Load Balancer controller](https://github.com/kubernetes-sigs/aws-load-balancer-controller), see [installation instructions](https://kubernetes-sigs.github.io/aws-load-balancer-controller/latest/deploy/installation/). Note that the repository maintains a webpage for each release. Confirm that you are using the latest installation instructions.
```

File: doc/source/cluster/kubernetes/k8s-ecosystem/ingress.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/k8s-ecosystem/ingress.md:17]:
```suggestion
* (Optional) Try the [echo server example](https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/main/docs/examples/echo_server.md) in the [aws-load-balancer-controller](https://github.com/kubernetes-sigs/aws-load-balancer-controller) repository.
```

File: doc/source/cluster/kubernetes/k8s-ecosystem/ingress.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/k8s-ecosystem/ingress.md:19]:
```suggestion
* (Optional) Read [how-it-works.md](https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/main/docs/how-it-works.md) to understand the [aws-load-balancer-controller](https://github.com/kubernetes-sigs/aws-load-balancer-controller) mechanism.
```

File: doc/source/cluster/kubernetes/k8s-ecosystem/prometheus-grafana.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/k8s-ecosystem/prometheus-grafana.md:3]:
```suggestion
# Using Prometheus and Grafana
```

File: doc/source/cluster/kubernetes/k8s-ecosystem/prometheus-grafana.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/k8s-ecosystem/prometheus-grafana.md:7]:
```suggestion
If you do not have any experience with Prometheus and Grafana on Kubernetes, watch this [YouTube playlist](https://youtube.com/playlist?list=PLy7NrYWoggjxCF3av5JKwyG7FFF9eLeL4).
```

File: doc/source/cluster/kubernetes/k8s-ecosystem/prometheus-grafana.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/k8s-ecosystem/prometheus-grafana.md:11]:
```suggestion
Clone the [KubeRay repository](https://github.com/ray-project/kuberay) and checkout the `master` branch.
```

File: doc/source/cluster/kubernetes/k8s-ecosystem/prometheus-grafana.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/k8s-ecosystem/prometheus-grafana.md:14]:
```suggestion
## Step 1: Create a Kubernetes cluster with Kind.
```
```suggestion
## Step 1: Create a Kubernetes cluster with Kind
```

File: doc/source/cluster/kubernetes/k8s-ecosystem/prometheus-grafana.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/k8s-ecosystem/prometheus-grafana.md:84]:
```suggestion
* KubeRay exposes a Prometheus metrics endpoint in port **8080** via a built-in exporter by default. Hence, we do not need to install any external exporter.
```

File: doc/source/cluster/kubernetes/k8s-ecosystem/pyspy.md
Comment (angelinalg) [inline doc/source/cluster/kubernetes/k8s-ecosystem/pyspy.md:70]:
```suggestion
- For additional details on using the profiler, See [Python CPU profiling in the Dashboard](https://docs.ray.io/en/latest/ray-observability/user-guides/debug-apps/optimize-performance.html#python-cpu-profiling-in-the-dashboard).
```

**Review by angelinalg — APPROVED** [review]
Spot check of the structure looks really good. Only had time for a few copy edits. Should follow up with more. Great job, @kevin85421 !

**Comment by angelinalg** [comment]
Not being able to create child pages under Managed Kubernetes does not surprise me. Max has had to do some hacks to get around long build times that are related to large side navs, or he might've introduced limits to clean up the side nav bloat.

Re: doc tests - when you build locally, you should be running the doc tests and the CI also runs the full suite of doc tests. You don't need to do anything else.

**Comment by angelinalg** [comment]
@kevin85421 , thank you for migrating the IPYNB file to MD!

File: doc/source/_toc.yml
Comment (kevin85421) [reply doc/source/_toc.yml:322]: Good catch!

Comment (kevin85421) [reply doc/source/_toc.yml:322]: Updated https://github.com/ray-project/ray/pull/38647/commits/22730ddb7a4c5abd6c5db7d935a3cb99f0aa677e.

File: doc/source/cluster/kubernetes/examples/mobilenet-rayservice.md
Comment (kevin85421) [reply doc/source/cluster/kubernetes/examples/mobilenet-rayservice.md:37]: Serve service here refers to Kubernetes service. Using Ray Serve service here may cause users to be confused.

File: doc/source/cluster/kubernetes/getting-started/rayjob-quick-start.md
Comment (kevin85421) [reply doc/source/cluster/kubernetes/getting-started/rayjob-quick-start.md]: Updated https://github.com/ray-project/ray/pull/38647/commits/21dcebace5896289dcfbe067c7dae54792a0060e

File: doc/source/cluster/kubernetes/examples/mobilenet-rayservice.md
Comment (kevin85421) [reply doc/source/cluster/kubernetes/examples/mobilenet-rayservice.md:19]: Unless you feel strongly about dividing it into two sections, I'd rather keep it as it is.

File: doc/source/cluster/kubernetes/examples/mobilenet-rayservice.md
Comment (kevin85421) [inline doc/source/cluster/kubernetes/examples/mobilenet-rayservice.md:37]:
```suggestion
Note that the Serve service is created after the Ray Serve applications are ready and running. This process may take approximately 1 minute after all Pods in the RayCluster are running.
```

File: doc/source/cluster/kubernetes/examples/mobilenet-rayservice.md
Comment (kevin85421) [reply doc/source/cluster/kubernetes/examples/mobilenet-rayservice.md:37]: https://github.com/ray-project/ray/pull/38647/files#r1303623134

**Review by zhe-thoughts — APPROVED** [review]
Yes this qualifies for merging. 

@architkulkarni could you check and give a green light of `tests-ok`? I'll wait for the green light before merging

**Comment by architkulkarni** [comment]
Failed tests:
```


//python/ray/serve:test_http_headers                                    TIMEOUT in 3 out of 3 in 60.1s
--
  | Stats over 3 runs: max = 60.1s, min = 60.1s, avg = 60.1s, dev = 0.0s
  | C:/tmp/4lhdprva/execroot/com_github_ray_project_ray/bazel-out/x64_windows-opt/testlogs/python/ray/serve/test_http_headers/test.log
  | C:/tmp/4lhdprva/execroot/com_github_ray_project_ray/bazel-out/x64_windows-opt/testlogs/python/ray/serve/test_http_headers/test_attempts/attempt_1.log
  | C:/tmp/4lhdprva/execroot/com_github_ray_project_ray/bazel-out/x64_windows-opt/testlogs/python/ray/serve/test_http_headers/test_attempts/attempt_2.log
  | //python/ray/tests:test_cancel                                          TIMEOUT in 3 out of 3 in 300.1s
  | Stats over 3 runs: max = 300.1s, min = 300.1s, avg = 300.1s, dev = 0.0s
  | C:/tmp/4lhdprva/execroot/com_github_ray_project_ray/bazel-out/x64_windows-opt/testlogs/python/ray/tests/test_cancel/test.log
  | C:/tmp/4lhdprva/execroot/com_github_ray_project_ray/bazel-out/x64_windows-opt/testlogs/python/ray/tests/test_cancel/test_attempts/attempt_1.log
  | C:/tmp/4lhdprva/execroot/com_github_ray_project_ray/bazel-out/x64_windows-opt/testlogs/python/ray/tests/test_cancel/test_attempts/attempt_2.log
  | //python/ray/serve:test_autoscaling_policy                               FAILED in 3 out of 3 in 66.9s
  | Stats over 3 runs: max = 66.9s, min = 65.4s, avg = 66.2s, dev = 0.6s
  | C:/tmp/4lhdprva/execroot/com_github_ray_project_ray/bazel-out/x64_windows-opt/testlogs/python/ray/serve/test_autoscaling_policy/test.log
  | C:/tmp/4lhdprva/execroot/com_github_ray_project_ray/bazel-out/x64_windows-opt/testlogs/python/ray/serve/test_autoscaling_policy/test_attempts/attempt_1.log
  | C:/tmp/4lhdprva/execroot/com_github_ray_project_ray/bazel-out/x64_windows-opt/testlogs/python/ray/serve/test_autoscaling_policy/test_attempts/attempt_2.log


```

Unrelated, this is a docs-only change

**Comment by kevin85421** [comment]
Hi @zhe-thoughts, I added a new commit https://github.com/ray-project/ray/pull/38647/commits/ec19d1556f06587f82b7d554ad71cfc6cfda566a, and I will let you know when the test finishes. Thanks!

**Comment by architkulkarni** [comment]
@kevin85421 when you're done updating the PR and CI is finished, you can go through the failed tests and add the tests-ok lael

**Comment by kevin85421** [comment]
`docs/readthedocs.com:anyscale-ray` builds successfully. cc @architkulkarni @zhe-thoughts

**Comment by zhe-thoughts** [comment]
Since this is a big change, @architkulkarni could you make a pass of tests and apply `tests-ok` after that? Thanks

**Comment by zhe-thoughts** [comment]
@kevin85421 sorry, need a rebase. There's conflict now

**Comment by kevin85421** [comment]
@zhe-thoughts I encountered no conflicts when rebasing with the upstream master branch locally. I still went ahead and rebased, then force-pushed. Thanks!

**Pull request merged**
